### ISSUE TITLE
Null pointer exception trying to test a rule

### ISSUE THREAD

**Issue by craigmcintyre** [antlr/intellij-plugin-v4#145]
Hi,

Environment: IntelliJ IDEA 13.1.4
ANTLR4 Plugin: latest version.
OS: Mac OSX 10.9.4
Java JDK: 1.7.0_55

I got the following output when trying to test a rule by right clicking on it in the IDE and selecting "Test Rule". It now does this for every rule in my project.

This started happening after the following sequence of events. I was testing various rules and viewing the parse trees when I decided to revert to a previous version of my grammar file from git. I did this from within IntelliJ and then tried to test the same rule from the older version of the grammar. At that point whenever I select Test Rule I get this error and the preview panel shows me an incomplete rendering as there is no input box to enter test data with (see image). I tried going back to the latest version of all my files but I now get this problem regardless. It would seem that something with my environment has become corrupted and it is unable to recover.

I have also tried uninstalling the ANTLR4 plugin from IntelliJ and reinstalling it. I even went so far as to reboot my Mac, but that had no effect either. It all previously worked fine until I decided to revert a file...

Thanks for your time.
Craig.

null
java.lang.NullPointerException
	at org.antlr.intellij.plugin.preview.PreviewPanel.updateParseTreeFromDoc(PreviewPanel.java:246)
	at org.antlr.intellij.plugin.ANTLRv4PluginController.setStartRuleNameEvent(ANTLRv4PluginController.java:251)
	at org.antlr.intellij.plugin.actions.TestRuleAction.actionPerformed(TestRuleAction.java:82)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:164)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter$1.run(ActionMenuItem.java:266)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:926)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:124)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:236)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2028)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:105)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:512)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:44)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:532)
	at java.awt.Component.processMouseEvent(Component.java:6414)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3275)
	at java.awt.Component.processEvent(Component.java:6179)
	at java.awt.Container.processEvent(Container.java:2084)
	at java.awt.Component.dispatchEventImpl(Component.java:4776)
	at java.awt.Container.dispatchEventImpl(Container.java:2142)
	at java.awt.Component.dispatchEvent(Component.java:4604)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4618)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4279)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4209)
	at java.awt.Container.dispatchEventImpl(Container.java:2128)
	at java.awt.Window.dispatchEventImpl(Window.java:2492)
	at java.awt.Component.dispatchEvent(Component.java:4604)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:717)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:690)
	at java.awt.EventQueue$3.run(EventQueue.java:688)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:687)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:697)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:520)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:335)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)

![image](https://cloud.githubusercontent.com/assets/8200683/3950366/1ea277d8-26c4-11e4-96c9-95d8c0b5d38c.png)

**Comment by parrt** [comment]
Hiya. Can you try with latest plugin?

**Comment by jrguenther** [comment]
Latest plugin works most of the time. Periodically, I get another NPE when testing a rule with the right-click menu:

```
java.lang.NullPointerException
	at org.antlr.intellij.plugin.preview.PreviewPanel.updateParseTreeFromDoc(PreviewPanel.java:246)
	at org.antlr.intellij.plugin.ANTLRv4PluginController.setStartRuleNameEvent(ANTLRv4PluginController.java:251)
	at org.antlr.intellij.plugin.actions.TestRuleAction.actionPerformed(TestRuleAction.java:82)
...
```

**Comment by parrt** [comment]
yeah. i’m working on that one.

**Comment by craigmcintyre** [comment]
Do you have a precompiled binary of the latest plugin available? I seem to be hitting dead ends trying to compile it up myself.

**Comment by parrt** [comment]
yup. posted to discussion group, but here ya go. http://antlr.org/download/intellij-plugin-v4-1.7b1.zip

**Comment by craigmcintyre** [comment]
Fantastic, thanks. I tried it out and so far I'm not getting the errors I was before. The only hiccup I ran into was that the plugin needs IntelliJ running under at least Java 7. It seems the default for IntelliJ under OSX is Java 6, even when Java 7 is installed (instructions [here](https://intellij-support.jetbrains.com/entries/23455956-Selecting-the-JDK-version-the-IDE-will-run-under) for modifying this).  

Thanks for the fix!
